de-lib/src/main/java/org/iplantc/de/diskResource/client/NavigationView.java

```diff
@@ -115,7 +113,15 @@
          */
         boolean isLoaded(Folder folder);
 
-        void refreshFolder(Folder folder);
+        boolean isPathUnderKnownRoot(String path);
+
+        /**
+         * Reloads the child folders under the given Folder for this view's TreeStore only.
+         * May trigger a re-selection of the currently selected folder.
```

Comment:
It's not always to trigger lazy-loading. If the selected folder is the refreshed folder, we just re-select it to reload the center grid.